### ISSUE TITLE
Feature: Ability to decode error response

### ISSUE THREAD

**Issue by malonehedges** [joshuawright11/papyrus#17]
I'd love to be able to decode an error response, but from what I can see the only way to do this is to opt-out of any built-in response decoding and handle the `Response` myself entirely.

**Comment by joshuawright11** [comment]
Thanks for the feedback @malonehedges! 

What's the use case you're thinking of?

Using the body from the raw `Response` is currently the only way. The error might not have a body (i.e. if there were a transport error) and having a custom error type on an unsuccessful status code would be tricky to have it built in. You'd need to have two types in the return; an expected `Res` type plus the possible `Err` type to decode.

You are able to use the build in `decode(...)` function on `Response` however:

```swift
@GET("/error")
func getError() async throws -> Response

let res = try await api.getError()
if let statusCode = res.statusCode, (200...399).contains(statusCode) {
    let errorBody = try res.decode(MyCustomErrorResponse)
} else {
    // handle success
}
```

**Comment by malonehedges** [comment]
In my APIs I typically return either 200 with the expected response type, or some error code and `{ "error": true, "message": "Handle must be between 4 and 16 characters" }` for example, so that I can show that message in the client

**Comment by malonehedges** [comment]
Ultimate goal is to turn a fetch failure into a message, either defined by the API in the response or falling back to "Something went wrong, please try again"

**Comment by F0x1d** [comment]
I can relate. Now we have to completely manually realize `Response` parsing if we want to get the body of error response. I suggest passing `ErrorResponse` with `PapyrusError` as it is done in `Retrofit`

**Comment by ucomah** [comment]
For the cases I met, used an additional interceptor which decodes specific API error responses. If it fails - default error processor will be used.

**Comment by joshuawright11** [comment]
Thanks @F0x1d for [the PR](#32) 💯 

This is released as of v0.6.2.
